WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -298,13 +297,40 @@ - (UINavigationController *)notificationsNavigationController
     return _notificationsNavigationController;
 }
 
+- (void)updateWriteButtonAppearance
+{
+    CGSize size = self.view.bounds.size;
+    CGFloat screenWidth = [[UIScreen mainScreen] bounds].size.width;
+
+    // Try and determine whether the app is displayed at a size which will result in a tab
+    // bar with button titles and images horizontally stacked, instead of vertically
+    BOOL iPhoneLandscape = [WPDeviceIdentification isiPhone] && UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation]);
+    BOOL iPadPortraitFullscreen = [WPDeviceIdentification isiPad] &&
+    UIInterfaceOrientationIsPortrait([[UIApplication sharedApplication] statusBarOrientation]) &&
+    size.width == screenWidth;
+    BOOL iPadLandscapeGreaterThanHalfSplit = [WPDeviceIdentification isiPad] &&
+    UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation]) &&
+    size.width > screenWidth / 2;
+
+    if (iPhoneLandscape || iPadPortraitFullscreen || iPadLandscapeGreaterThanHalfSplit) {
+        self.newPostViewController.tabBarItem.imageInsets = UIEdgeInsetsZero;
+        self.newPostViewController.tabBarItem.titlePositionAdjustment = UIOffsetZero;
+        self.newPostViewController.tabBarItem.image = [Gridicon iconOfType:GridiconTypeCreate withSize:CGSizeMake(WPTabBarIconSize, WPTabBarIconSize)];
+    } else {
+        self.newPostViewController.tabBarItem.imageInsets = [self tabBarIconImageInsets];
+        self.newPostViewController.tabBarItem.titlePositionAdjustment = UIOffsetMake(0, 99999.0);
```

Comment:
I tested and another option to avoid this ugly offset is to set the title to nil here and to Write on the if, something like:
```
if (iPhoneLandscape || iPadPortraitFullscreen || iPadLandscapeGreaterThanHalfSplit) {
        self.newPostViewController.tabBarItem.imageInsets = UIEdgeInsetsZero;
        self.newPostViewController.tabBarItem.titlePositionAdjustment = UIOffsetZero;
        self.newPostViewController.tabBarItem.image = [Gridicon iconOfType:GridiconTypeCreate withSize:CGSizeMake(WPTabBarIconSize, WPTabBarIconSize)];
        self.newPostViewController.tabBarItem.title = NSLocalizedString(@"Write", @"The accessibility value of the post tab.");
} else {
        self.newPostViewController.tabBarItem.imageInsets = [self tabBarIconImageInsets];
        self.newPostViewController.tabBarItem.title = nil;

        UIImage *newPostImage = [UIImage imageNamed:@"icon-tab-newpost"];
        newPostImage = [newPostImage imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
        self.newPostViewController.tabBarItem.image = newPostImage;
 }
```

I don't feel strongly about it though, what do you think?